R/update_na_vals.R

```diff
@@ -60,20 +60,24 @@ update_na_values <- function(dt,
 
     if (is_data_table) {
 
-      dt_1[get(reportvar) == 4,
-         (x.var) := mget(y.var)]
-
-      dt_1[get(reportvar) == 5,
-         eval(x.var) := mget(y.var)]
+      dt_1[dt_1[[reportvar]] == 4,
+           x.var] <- dt_1[dt_1[[reportvar]] == 4,
+                          y.var,
+                          with = FALSE]
+#
+      dt_1[dt_1[[reportvar]] == 5,
+           x.var] <- dt_1[dt_1[[reportvar]] == 5,
+                          y.var,
+                          with = FALSE]
```

Comment:
Hi @zander-prinsloo , I don't understand this change. It is now making a deep copy rather than modifying by reference. What is the purpose of this change?

Reply:
@randrescastaneda , previously it was making a deep copy already using `copy()` and then modifying by reference after that. But you are right, this is not a sufficient improvement on what was done before. Let me propose a modification.